lld/ELF/SyntheticSections.cpp

```diff
@@ -2760,20 +2767,21 @@ void DebugNamesBaseSection::parseDebugNames(
     const uint32_t hdrSize =
         getDebugNamesHeaderSize(nd.hdr.AugmentationStringSize);
     auto locs = findDebugNamesOffsets(hdrSize, nd.hdr);
```

Comment:
I don't think we need to call this function, or add/track `niOffset` at all (whatever we do, we shouldn't try to rebuild the name index. table offsets - that seems likely to cause bugs if this code ever rebuilt it incorrectly/inconsistently with the way the indexes were parsed in the first place)

instead, I think `ni.Offsets` already contains the offsets we want here (already adjusted for/including the offset where the debug_names table starts witihn the .debug_names section), right? So if there was an accessor for that added to `DWARFDebugNames::NameIndex`, we wouldn't need to be recomputing the offsets, etc here?

Reply:
Done.